src/writer.rs

```diff
@@ -261,6 +265,33 @@ impl<W: Write> Writer<W> {
             start_tag: BytesStart::new(name.as_ref()),
         }
     }
+
+    /// Write an arbitrary serializable object
+    #[cfg(feature = "serialize")]
+    pub fn write_serializable<T: Serialize>(
+        &mut self,
+        content: &T,
+        tag_name: &str,
```

Comment:
In theory we could make this optional very easily - passing `None` would then use the struct name by default.  On review however I wasn't sure that it would actually be useful.

Reply:
Yes, I think it would be better to always require to specify tag name explicitly. Maybe also swap `tag_name` and `content`? The
```rust
writer.write_serializable("key", &value)?;
```
looks more natural than
```rust
writer.write_serializable(&value, "key")?;
```